crates/services/compression/src/ports/block_source.rs

```diff
@@ -0,0 +1,11 @@
+/// Type alias for fuel_core_types::blockchain::block::Block
+pub type Block = fuel_core_types::blockchain::block::Block;
+
+/// Port for L2 blocks source
+/// Each time `.next()` is called, the service expects the same behaviour as an iterator.
+pub trait BlockSource {
+    /// Get the next block from the source
+    fn next(
+        &self,
+    ) -> impl core::future::Future<Output = crate::Result<Option<Block>>> + Send + Sync;
+}
```

Comment:
Nit: I'd rename the method to `next_block` for clarity. But I think `next` probably also works if the variable in the calling code is properly named.
```suggestion
pub type Block = fuel_core_types::blockchain::block::Block;

/// Port for L2 blocks source
/// Each time `.next()` is called, the service expects the same behaviour as an iterator.
pub trait BlockSource {
    /// Get the next block from the source
    fn next_block(
        &self,
    ) -> impl core::future::Future<Output = crate::Result<Option<Block>>> + Send + Sync;
}
```

Reply:
block_source.next() makes more sense to me than block_source.next_block(). also given that we allude to an iterator format here I think it is reasonable to assume the stream is of blocks.

Reply:
Fair enough. I agree that `block_source.next()` sounds best. However I'd rather see `block_source.next_block()` rather than `source.next()`, which feels like it can pop up in client code. But that's just a matter of taste, and I like the conciseness of just calling it `.next()`. Perhaps that will encourage better variable names as well.